squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/impl/FrontierImpl.java

```diff
@@ -298,36 +274,41 @@ public void addNewUri(CrawleableUri uri) {
                 knownUriFilter.add(uri, System.currentTimeMillis());
             } else {
                 LOGGER.warn("addNewUri(" + uri + "): " + uri.getUri().getScheme() + " is not supported, only "
-                        + schemeUriFilter.getSchemes() + ". Will not added!");
+                    + schemeUriFilter.getSchemes() + ". Will not added!");
             }
 
         } else {
             LOGGER.debug("addNewUri(" + uri + "): URI is not good [" + knownUriFilter + "]. Will not be added!");
         }
     }
 
-
     @Override
     public void crawlingDone(List<CrawleableUri> uris) {
         LOGGER.info("One worker finished his work and crawled " + uris.size() + " URIs.");
-
-        // List<CrawleableUri> newUris = new ArrayList<>(uriMap.size());
-        // for (CrawleableUri uri : uriMap.keySet()) {
-        // newUris.addAll(uriMap.get(uri));
-        // knownUriFilter.add(uri, System.currentTimeMillis(),
-        // uri.getTimestampNextCrawl());
-        // if (uriReferences != null) {
-        // uriReferences.add(uri, uriMap.get(uri));
-        // }
-        // }
-
-        // // If there is a graph logger, log the data
-        // if (graphLogger != null) {
-        // graphLogger.log(new ArrayList<>(uriMap.keySet()), newUris);
-        // }
+//        List<CrawleableUri> newUris = new ArrayList<>(uriMap.size());
+//        for (CrawleableUri uri : uriMap.keySet()) {
+//            newUris.addAll(uriMap.get(uri));
+//            knownUriFilter.add(uri, System.currentTimeMillis(), uri.getTimestampNextCrawl());
+//            if (uriReferences != null) {
+//                uriReferences.add(uri, uriMap.get(uri));
+//            }
+//        }
+
+//        // If there is a graph logger, log the data
+//        if (graphLogger != null) {
+//            graphLogger.log(new ArrayList<>(uriMap.keySet()), newUris);
+//        }
         // If we should give the crawled IPs to the queue
-        if (queue instanceof BlockingQueue) {
-            ((BlockingQueue<?>) queue).markUrisAsAccessible(uris);
+        if (queue instanceof IpAddressBasedQueue) {
+            Set<InetAddress> ips = new HashSet<>();
+            InetAddress ip;
+            for (CrawleableUri uri : uris) {
+                ip = uri.getIpAddress();
+                if (ip != null) {
+                    ips.add(ip);
+                }
+            }
+            ips.forEach(_ip -> ((IpAddressBasedQueue) queue).markIpAddressAsAccessible(_ip));
```

Comment:
The changes you introduce look wrong. I am afraid you are reintroducing an outdated handling of URIs. Please check and fix it.